### PR TITLE
Feat(eos_designs): Adding schemas for eos_designs missing keys

### LINKED ISSUE #2834: Add schemas for missing keys in eos_designs
**eos_designs:**

- [ ] Add schema for `evpn_ebgp_gateway_inter_domain` #2862
- [ ] Add schema for `isis_maximum_paths` #2862
- [ ] Add schema for `avd_data_conversion_mode` #2862
- [ ] Add schema for `avd_data_validation_mode` #2862
- [ ] Add schema for `eos_designs_documentation` #2862
- [ ] Add schema for `avd_requirements`
- [ ] Add schema for `mgmt_interface_description` - @chetryan #2858 
- [ ] Add schema for `ipv6_mgmt_gateway` #2862
- [ ] Add schema for `ipv6_mgmt_destination_networks` #2862
- [ ] Add schema for `overlay_cvx_servers` #2862
- [ ] Add schema for `serial_number` #2862
- [ ] Add schema for `underlay_filter_redistribute_connected` #2862
- [ ] Add schema for `is_deployed`.   - @chetryan #2858 
- [ ] Add schema for `fabric_evpn_encapsulation` - shivani

**eos_cli_config_gen:**

- [ ] Add schema for `serial_number`
- [ ] Add schema for `eos_cli_config_gen_configuration`
- [ ] Add schema for `eos_cli_config_gen_documentation`

Remove:

- [ ] `vxlan_vlan_aware_bundles` in favor of `evpn_vlan_aware_bundles` #2865 

Role inputs - can be postponed to 4.x

```yaml
  root_dir:
  documentation_dir_name:
  documentation_dir:
  devices_dir_name:
  devices_dir:
  output_dir_name:
  output_dir:
  structured_dir_name:
  structured_dir:
  eos_config_dir_name:
  eos_config_dir:
  fabric_dir_name:
    type: str
  fabric_dir:
    type: str

eos_config_deploy_eapi:
  eos_config_deploy_eapi_pre_running_config_backup:
    type: any
  post_running_config_backup_filename:
    type: any
  post_running_config_backup_dir_name:
    type: any
  post_running_config_backup_dir:
    type: any
  pre_running_config_backup_filename:
    type: any
  pre_running_config_backup_dir_name:
    type: any
  pre_running_config_backup_dir:
    type: any
```

---
Full list for reference:
```yaml
  root_dir:
    type: str
  documentation_dir_name:
    type: str
  documentation_dir:
    type: str
  devices_dir_name:
    type: str
  devices_dir:
    type: str
  output_dir_name:
    type: str
  output_dir:
    type: str
  structured_dir_name:
    type: str
  structured_dir:
    type: str
  eos_config_dir_name:
    type: str
  eos_config_dir:
    type: str
  avd_structured_config_file_format:
    type: str
  molecule_ephemeral_directory:
    type: any
  molecule_file:
    type: any
  molecule_instance_config:
    type: any
  molecule_no_log:
    type: any
  molecule_scenario_directory:
    type: any
  molecule_yml:
    type: str
  is_for_test:
    type: bool
  inventory_file:
    type: str
  inventory_dir:
    type: str
  inventory_hostname:
    type: str
  inventory_hostname_short:
    type: str
  group_names:
    type: list
  ansible_facts:
    type: any
  playbook_dir:
    type: any
  ansible_playbook_python:
    type: any
  ansible_config_file:
    type: any
  groups:
    type: any
  omit:
    type: any
  ansible_version:
    type: any
  ansible_check_mode:
    type: any
  ansible_diff_mode:
    type: any
  ansible_forks:
    type: any
  ansible_inventory_sources:
    type: any
  ansible_skip_tags:
    type: any
  ansible_limit:
    type: any
  ansible_run_tags:
    type: any
  ansible_verbosity:
    type: any
  eosconfig:
    type: dict
  default_connected_endpoints_keys:
    type: dict
  default_node_type_keys:
    type: dict
  default_platform_settings:
    type: list
  default_templates:
    type: dict
  fabric_dir_name:
    type: str
  fabric_dir:
    type: str
  evpn_ebgp_gateway_inter_domain:
    type: bool
  isis_maximum_paths:
    type: int
  custom_structured_configuration_prefix:
    type: list
    convert_types:
      - str
  custom_structured_configuration_list_merge:
    type: str
  templates:
    type: dict
  avd_data_conversion_mode:
    type: str
  avd_data_validation_mode:
    type: str
  ptp_profiles:
    type: list
  eos_designs_documentation:
    type: dict
  avd_switch_facts:
    type: dict
  avd_overlay_peers:
    type: dict
  avd_topology_peers:
    type: dict
  role_schema_dir_name:
    type: str
  role_schema_dir:
    type: str
  role_schema_file_name:
    type: str
  role_schema_path:
    type: str
  role_schema_fragments_dir_name:
    type: str
  role_schema_fragments_dir:
    type: str
  role_schema_fragments_file_glob:
    type: str
  role_schema_fragments_glob:
    type: str
  role_schema_json_schema_file_name:
    type: str
  role_schema_json_schema_path:
    type: str
  structured_config_schema_path:
    type: str
  avd_requirements:
    type: dict
  ansible_role_names:
    type: any
  ansible_play_role_names:
    type: any
  ansible_dependent_role_names:
    type: any
  role_names:
    type: any
  ansible_play_name:
    type: any
  role_name:
    type: any
  role_path:
    type: any
  role_uuid:
    type: any
  ansible_collection_name:
    type: any
  ansible_role_name:
    type: any
  ansible_play_hosts_all:
    type: any
  ansible_play_hosts:
    type: any
  ansible_play_batch:
    type: any
  play_hosts:
    type: any
  hostvars:
    type: any
  environment:
    type: any
  vars:
    type: any
  ansible_delegated_vars:
    type: any
  ansible_current_hosts:
    type: any
  ansible_failed_hosts:
    type: any
  ansible_search_path:
    type: any
  ansible_timeout:
    type: any
  ansible_shell_executable:
    type: any
  ansible_host:
    type: any
  ansible_connection:
    type: any
  ansible_module_compression:
    type: any
  ansible_pipelining:
    type: any
  ansible_shell_type:
    type: any
  switch:
    type: dict
  hardware_counters:
    type: dict
  ntp:
    type: dict
  spine_platform:
    type: any
  spine_bgp_defaults:
    type: any
  boot:
    type: any
  tenant_a:
    type: any
  tenant_b:
    type: any
  tenant_c:
    type: any
  tenant_d:
    type: any
  mgmt_interface_description:
    type: str
  ipv6_mgmt_gateway:
    type: str
  ipv6_mgmt_destination_networks:
    type: list
  overlay_cvx_servers:
    type: any
  serial_number:
    type: any
  underlay_filter_redistribute_connected:
    type: bool
  ip_offset_10:
    type: any
  ip_offset_20:
    type: any
  description_prefix:
    type: any
  my_dci_ethernet_interfaces:
    type: any
  override_ntp:
    type: any
  my_special_dci_ethernet_interfaces:
    type: any
  override_ip_name_servers:
    type: any
  template_timezone:
    type: any
  my_mac_address_table:
    type: any
  override_mac_address_table:
    type: any
  is_deployed:
    type: any
  ansible_httpapi_port:
    type: any
  ansible_httpapi_host:
    type: any
  ansible_httpapi_use_ssl:
    type: any
  ansible_httpapi_validate_certs:
    type: any
  ansible_network_os:
    type: any
  ansible_user:
    type: any
  ansible_ssh_pass:
    type: any
  ansible_become:
    type: any
  ansible_become_method:
    type: any
  ansible_port:
    type: any
  fabric_evpn_encapsulation:
    type: any
  queue_monitor_length:
    type: any
  vxlan_vlan_aware_bundles:
    type: any
  eos_cli_config_gen_configuration:
    type: any
  eos_cli_config_gen_documentation:
    type: any
  custom_structured_configuration_router_general:
    type: any
  aaa_authorization:
    type: any
  ansible_password:
    type: any
  role_documentation_dir_name:
    type: any
  role_documentation_dir:
    type: any
  aaa_authentication:
    type: any
  eos_config_deploy_eapi_pre_running_config_backup:
    type: any
  post_running_config_backup_filename:
    type: any
  post_running_config_backup_dir_name:
    type: any
  post_running_config_backup_dir:
    type: any
  pre_running_config_backup_filename:
    type: any
  pre_running_config_backup_dir_name:
    type: any
  pre_running_config_backup_dir:
    type: any
```

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Related Issue(s)

Fixes #2834 

## Component(s) name

`arista.avd.eos_designs`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/eos_designs_documentation.schema.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      filename: Fabric Variables
+    description: |
+      Some variables are available to control documentation generation. This can be useful
+      for instance the project has thousands of endpoints, to be able to disable
+      fabric-wide connected endpoints documentation.
+    type: dict
+    keys:
+      connected_endpoints:
+        type: bool
+        default: "true"
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/eos_designs_documentation.schema.yml]:
```suggestion
  eos_designs_documentation:
    documentation_options:
      filename: Fabric Variables
    description: |
      Control fabric documentation generation.
    type: dict
    keys:
      connected_endpoints:
        description: |
          Generate fabric-wide documentation for connected endpoints.
        type: bool
        default: false
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/underlay_filter_redistribute_connected.schema.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  underlay_filter_redistribute_connected:
+    documentation_options:
+      filename: Fabric Variables
+      table: BGP Settings
+    description: |
+      Filter redistribution of connected into underlay routing protocol.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/underlay_filter_redistribute_connected.schema.yml]:
```suggestion
      Filter redistribution of connected into the underlay routing protocol.
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/ipv6_mgmt_gateway.schema.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  ipv6_mgmt_gateway:
+    documentation_options:
+      filename: Management Settings
+    type: str
+    format: ipv6
+    description: IPv6 address
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/ipv6_mgmt_gateway.schema.yml]:
```suggestion
    description: |
      OOB Management interface gateway in IPv6 format.
      Used as next-hop for default gateway or static routes defined under 'ipv6_mgmt_destination_networks'

```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/ipv6_mgmt_destination_networks.schema.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  ipv6_mgmt_destination_networks:
+    documentation_options:
+      filename: Management Settings
+    type: list
+    items:
+      type: str
+    description: IPv6_network/Mask
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/ipv6_mgmt_destination_networks.schema.yml]:
```suggestion
  ipv6_mgmt_destination_networks:
    documentation_options:
      filename: Management Settings
    description: |
      List of IPv6 prefixes to configure as static routes towards the OOB IPv6 Management interface gateway.
      Replaces the default route.
    type: list
    items:
      type: str
      description: IPv6_network/Mask
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/avd_data_conversion_mode.schema.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  avd_data_conversion_mode:
+    type: str
+    valid_values: ["disabled", "error", "warning", "info", "debug", "quiet"]
+    default: "debug"
+    description: |
+       Conversion Mode for AVD input data conversion
+       During conversion, messages will be generated with information about the host(s) and key(s) which required conversion.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/avd_data_conversion_mode.schema.yml]:
```suggestion
       Conversion Mode for AVD input data conversion
       Input data conversion will perform type conversion of input variables as defined in the schema.
       The type conversion is intended to help the user to identify minor issues with the input data, while still allowing the data to be validated.
       During conversion, messages will generated with information about the host(s) and key(s) which required conversion.
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/avd_data_validation_mode.schema.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  avd_data_validation_mode:
+    type: str
+    valid_values: ["disabled", "error", "warning", "info", "debug"]
+    default: "warning"
+    description: |
+       Validation Mode for AVD input data validation
+       During validation, messages will generated with information about the host(s) and key(s) which failed validation.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/avd_data_validation_mode.schema.yml:13]:
```suggestion
       Input data validation will validate the input variables according to the schema.
       During validation, messages will generated with information about the host(s) and key(s) which failed validation.
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/overlay_cvx_servers.schema.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  overlay_cvx_servers:
+    documentation_options:
+      filename: Fabric Variables
+    description: |
+      Overlay settings when using CVX as overlay controller.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/overlay_cvx_servers.schema.yml]:
```suggestion
      List of CVX vxlan overlay controllers.
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/overlay_cvx_servers.schema.yml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+      Required if overlay_routing_protocol == CVX.
+      CVX servers (VMs) are peering using their management interface, so mgmt_ip must be set for all CVX servers.
+    type: list
+    items:
+      type: str
+      description: CVX_inventory_hostname
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/overlay_cvx_servers.schema.yml]:
```suggestion
      description: 'inventory_hostname' of CVX server
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/serial_number.schema.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    documentation_options:
+      filename: Fabric Topology
+    description: |
+      Set to the Serial Number of the device
+      For now only used for documentation purpose in the fabric documentation.
+      "serial_number" can also be set directly as a hostvar.
+      If both are set, the setting under "Fabric Topology" takes precedence.
+    type: str
+    convert_types:
+      - int
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/serial_number.schema.yml]:
```suggestion
  serial_number:
    documentation_options:
      filename: Fabric Variables
    description: |
      Serial Number of the device
      Used for documentation purpose in the fabric documentation as can also be used by the 'eos_config_deploy_cvp' role.
      "serial_number" can also be set directly under "Fabric Topology".
      If both are set, the setting under "Fabric Topology" takes precedence.
    type: str
    convert_types:
      - int
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/underlay_filter_redistribute_connected.schema.yml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+keys:
+  underlay_filter_redistribute_connected:
+    documentation_options:
+      filename: Fabric Variables
+      table: BGP Settings
+    description: |
+      Filter redistribution of connected into the underlay routing protocol.
+      Only applicable when overlay_routing_protocol != 'none' and underlay_routing_protocol == BGP.
+      Creates a route-map and prefix-list assigned to redistribute connected permitting only loopbacks and inband management subnets.
+    default: "true"
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/underlay_filter_redistribute_connected.schema.yml]:
```suggestion
    default: true
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/avd_data_conversion_mode.schema.yml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  avd_data_conversion_mode:
+    type: str
+    valid_values: ["disabled", "error", "warning", "info", "debug", "quiet"]
+    default: "debug"
+    description: |
+       Conversion Mode for AVD input data conversion
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/avd_data_conversion_mode.schema.yml]:
```suggestion
       Conversion Mode for AVD input data conversion.
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/avd_data_validation_mode.schema.yml [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  avd_data_validation_mode:
+    type: str
+    valid_values: ["disabled", "error", "warning", "info", "debug"]
+    default: "warning"
+    description: |
+       Validation Mode for AVD input data validation
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/avd_data_validation_mode.schema.yml]:
```suggestion
       Validation Mode for AVD input data validation.
```

**Review by carlbuchmann — APPROVED** [review]
LGTM - minor issue to documentation, we need to add `"avd": "AVD"` to plugins/schema/key_to_display_name.py.

But we can do this in another PR where we will review the documentation!

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**